packages/storage/src/cookies.ts

```diff
@@ -89,45 +93,16 @@ try {
 export const cookieStorage: StorageWithOptions<CookieOptions> = addClearMethod({
   _read: isServer
     ? (options?: CookieOptions) => {
-        const eventOrRequest = options?.getRequest?.() || useRequest();
-        const request =
-          eventOrRequest && ("request" in eventOrRequest ? eventOrRequest.request : eventOrRequest);
-        let result = "";
-        if (eventOrRequest.responseHeaders) {
-          // Check if we really got a pageEvent
-          const responseHeaders = eventOrRequest.responseHeaders as Headers;
-          result +=
-            responseHeaders
-              .get("Set-Cookie")
-              ?.split(",")
-              .map(cookie => !cookie.match(/\\w*\\s*=\\s*[^;]+/))
-              .join(";") ?? "";
-        }
-        return `${result};${request?.headers?.get("Cookie") ?? ""}`; // because first cookie will be preferred we don't have to worry about duplicates
-      }
+      const headers = options?.getRequestHeaders?.() || getRequestHeaders();
```

Comment:
I'm appending cookies now, so that is no longer an issue.

Reply:
Not a fan of that, because the request header is in my opinion considered immutable, because the request didn't change and when i am querying the request header i want exactly the request real request headers and not the response ones

Reply:
But we're handling the response headers here, not the request headers.

Reply:
Ah, I misunderstood which code you were referring to. Now that is a problem. I will take a bit of time to think of a better solution.

Reply:
My solution was to concat the request header and the response header, so that the response header overwrites the values of the request header.

Reply:
I now read the request headers and then the response headers; the last value will overwrite previous ones, which should coincide with the behavior of most servers (unless you forcefully implement to do otherwise).